src/main/java/uk/gov/hmcts/reform/sscs/service/exceptions/FeignClientErrorDecoder.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.hmcts.reform.sscs.service.exceptions;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.Request;
+import feign.Request.HttpMethod;
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.hmcts.reform.sscs.model.HmcFailureMessage;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingRequestPayload;
+import uk.gov.hmcts.reform.sscs.service.AppInsightsService;
+
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+
+@Slf4j
+public class FeignClientErrorDecoder implements ErrorDecoder {
+
+    private final AppInsightsService appInsightsService;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    public FeignClientErrorDecoder(AppInsightsService appInsightsService) {
```

Comment:
As far as I know, it does not need to be autowired because it is a config class? Can you guys confirm? @Valantinas @cj1993 @dattipoeHMCTS 

https://stackoverflow.com/questions/32078600/why-do-i-not-need-autowired-on-bean-methods-in-a-spring-configuration-class

Reply:
@ThomasKKC Does ErrorDecoder have the @Configuration annotation on it?

Reply:
@dattipoeHMCTS Nope, but the FeignClientConfig class is. 
@FeignClient(name = "hmc-hearing", url = "${hmc.url}", configuration = FeignClientConfig.class)

Reply:
@ThomasKKC Ah right, I just saw that you're actually passing it down from FeignClientConfig!